docs/root/intro/arch_overview/http/http3.rst

```diff
@@ -18,42 +18,61 @@ HTTP3 downstream
 
-* :repo:`UDP listener downstream_rx_datagram_dropped </docs/root/configuration/listeners/stats.rst#udp-statistics>`: non-zero means kernel's UDP
-  listen socket's receive buffer isn't large enough. In Linux, it can be configured via listener :ref:`socket_options <envoy_v3_api_field_config.listener.v3.Listener.socket_options>`
-  by setting prebinding socket option SO_RCVBUF at SOL_SOCKET level.
-* :repo:`QUIC connection error codes and stream reset error codes </docs/root/configuration/http/http_conn_man/stats.rst#http3-per-listener-statistics>`: please
-  refer to `quic_error_codes.h <https://github.com/google/quiche/blob/main/quiche/quic/core/quic_error_codes.h>`_ for the meaning of each error code.
+:ref:`UDP listener downstream_rx_datagram_dropped <config_listener_stats_udp>`
+    Non-zero means kernel's UDP listen socket's receive buffer isn't large enough. In Linux,
+    it can be configured via listener :ref:`socket_options <envoy_v3_api_field_config.listener.v3.Listener.socket_options>`
+    by setting prebinding socket option ``SO_RCVBUF`` at ``SOL_SOCKET`` level.
+:repo:`QUIC connection error codes and stream reset error codes </docs/root/configuration/http/http_conn_man/stats.rst#http3-per-listener-statistics>`
+    Refer to `quic_error_codes.h <https://github.com/google/quiche/blob/main/quiche/quic/core/quic_error_codes.h>`_
+    for the meaning of each error code.
 
 HTTP3 upstream
 --------------
 
 HTTP/3 upstream support is implemented, with support both for explicit HTTP/3 (for data center use) and
-automatic HTTP/3 (for internet use). If you are in a controlled environment where UDP is unlikely to be blocked,
-you can configure it as the explicit protocol in
-:ref:`http_protocol_options <envoy_v3_api_msg_extensions.upstreams.http.v3.HttpProtocolOptions>`. For internet use,
-configuring auto_http with http3_protocol_options will result in Envoy attempting to use HTTP/3 for endpoints which
-have explicitly advertised HTTP/3 support via an ``alt-svc header``. Envoy will attempt to create a QUIC connection,
-then if the QUIC handshake is not complete after a short delay, will kick off a TCP connection, and will use whichever
-is established first.
-
-See :ref:`here <arch_overview_http3_upstream>` for more information about HTTP/3 connection pooling, including
-detailed information of where QUIC will be used, and how it fails over to TCP when QUIC use is configured to be optional.
-
-An example upstream HTTP/3 configuration file can be found :repo:`here </configs/google_com_http3_upstream_proxy.yaml>`.
+automatic HTTP/3 (for internet use).
+
+If you are in a controlled environment where UDP is unlikely to be blocked, you can configure it as the
+explicit protocol in :ref:`http_protocol_options <envoy_v3_api_msg_extensions.upstreams.http.v3.HttpProtocolOptions>`.
+
+For internet use, configuring :ref:`auto_config <envoy_v3_api_field_extensions.upstreams.http.v3.HttpProtocolOptions.auto_config>`
```

Comment:
@RyanTheOptimist @alyssawilk could either of you confirm this is correct?

i couldint find any reference to `auto_http` other than you guys discussing it

Reply:
This is accurate, but I think with the spacing is a bit less clear that the failover (line 71) only happens for auto_http and not for both cases.   IMo it should be in the same paragraph as auto_http and it doesn't look like it to me?

Reply:
what is `auto_http` can you point me to some docs?

Reply:
nm - read better your comment - `auto_http` -> `auto_config` 

ill check spacing now

Reply:
k - got it - i would like to keep the paragraph separate because i think it is a separate (but related) point - but ill prefix to make it clearer that its a continuation of previous paragraph

Reply:
yeah totally also fine to qualify "for auto_http .... blah"